external-crates/move/crates/move-stackless-bytecode/src/conditions_order_analysis.rs

```diff
@@ -0,0 +1,376 @@
+                                            call_node_id = Some(node);
+                                            call_operation = Some(operation.clone());
+                                        }
+                                    },
+                                    _ => {}
+                                };
+                            },
+                            _ => {},
+                        }
+                    }
+                },
+            };                
+        }
+
+        (call_node_id, call_operation, multiple)
+    }
+
+    pub fn fing_node_operation(&self, block_id: BlockId, cfg: &StacklessControlFlowGraph, code: &[Bytecode], targets: &[Operation], builder: &FunctionDataBuilder) -> Option<Loc> {
+        match cfg.content(block_id) {
+            BlockContent::Dummy => {},
+            BlockContent::Basic { lower, upper } => {
+                for position in *lower..*upper {
+                    match &code[position as usize] {
+                        Bytecode::Call(attr, _, opr, _, _) => {
+                            if targets.contains(opr) {
+                                return Some(builder.get_loc(*attr));
+                            }
+                        },
+                        _ => {},
+                    }
+                }
+            }
+        }
+    
+        return None;
+    }
+
+    pub fn fing_operations_before_after_operation_in_node(&self, block_id: &BlockId, operation: &Operation, cfg: &StacklessControlFlowGraph, code: &[Bytecode], builder: &FunctionDataBuilder, before_targets: &[Operation], after_targets: &[Operation]) -> (BTreeSet<Loc>, BTreeSet<Loc>) {
+        let mut befores = BTreeSet::new();
+        let mut afters = BTreeSet::new();
+        let mut matched = false;
+
+        match cfg.content(*block_id) {
+            BlockContent::Dummy => {},
+            BlockContent::Basic { lower, upper } => {
+                for position in *lower..*upper {
+                    match &code[position as usize] {
+                        Bytecode::Call(attr, _, opr, _, _) => {
+                            if opr == operation {
+                                matched = true;
+                            }
+
+                            if !matched && before_targets.contains(opr) {
+                                befores.insert(builder.get_loc(*attr));
+                            }
+
+                            if matched && after_targets.contains(opr) {
+                                afters.insert(builder.get_loc(*attr));
+                            }
+                        },
+                        _ => {},
+                    }
+                }
+            }
+        }
+    
+        return (befores, afters);
+    }
+    
+    pub fn find_modifiable_locations_in_graph(&self, graph: &Graph<BlockId>, code: &[Bytecode], cfg: &StacklessControlFlowGraph, builder: &FunctionDataBuilder, env: &GlobalEnv, skip: &Operation) -> BTreeSet<Loc> {
```

Comment:
this should be a recursive check, as a function may call a function that calls `transfer`